models/User.js

```diff
@@ -47,10 +47,19 @@ module.exports = (sequelize, DataTypes) => {
       allowNull: true,
     },
   };
-  const User = sequelize.define('User', userSchema);
+
+  const User = sequelize.define('User', userSchema, {
+    classMethods: {
+      associate(models) {
+        User.hasMany(models.Article);
+      }
+    }
+  });
   User.hook('beforeValidate', (user) => {
     user.verificationId = shortId.generate();
-    user.password = bcrypt.hashSync(user.password, bcrypt.genSaltSync(10));
+    if (user.password) {
```

Comment:
@kleva-j why do we have to check if a password was submitted?

Reply:
@9jaswag The `beforeValidate` hook runs for different methods like `User.update()`, and if it doesn't see the password field while trying to update the user's data, bcrypt will throw an error. So we decided to check if the password was supplied then hash it with bcrypt. If not, when a user wants to update his/her account, beforeValidate checks for password and if it's not provided, it'll not hash the password. So in this case, the password hashing happens only when a user signs up